code/game/gamemodes/wizard/soulstone.dm

```diff
@@ -263,55 +263,30 @@
-						Z.key = A.key
-						Z.faction |= "\ref[U]"
-						if(iscultist(U))
-							if(SSticker.mode.name == "cult")
-								SSticker.mode:add_cultist(Z.mind)
-							else
-								SSticker.mode.cult+=Z.mind
-							SSticker.mode.update_cult_icons_added(Z.mind)
-						qdel(T)
 						to_chat(Z, "<B>You are a Juggernaut. Though slow, your shell can withstand extreme punishment, create shield walls and even deflect energy weapons, and rip apart enemies and walls alike.</B>")
-						to_chat(Z, "<B>You are still bound to serve your creator, follow [U.p_their()] orders and help [U.p_them()] complete [U.p_their()] goals at all costs.</B>")
-						Z.cancel_camera()
-						qdel(C)
+						init_construct(Z,A,C,T)
 
 					if("Wraith")
 						var/mob/living/simple_animal/hostile/construct/wraith/Z = new /mob/living/simple_animal/hostile/construct/wraith (get_turf(T.loc))
-						Z.key = A.key
-						Z.faction |= "\ref[U]"
-						if(iscultist(U))
-							if(SSticker.mode.name == "cult")
-								SSticker.mode:add_cultist(Z.mind)
-							else
-								SSticker.mode.cult+=Z.mind
-							SSticker.mode.update_cult_icons_added(Z.mind)
-						qdel(T)
 						to_chat(Z, "<B>You are a Wraith. Though relatively fragile, you are fast, deadly, and even able to phase through walls.</B>")
-						to_chat(Z, "<B>You are still bound to serve your creator, follow [U.p_their()] orders and help [U.p_them()] complete [U.p_their()] goals at all costs.</B>")
-						Z.cancel_camera()
-						qdel(C)
+						init_construct(Z,A,C,T)
 
 					if("Artificer")
 						var/mob/living/simple_animal/hostile/construct/builder/Z = new /mob/living/simple_animal/hostile/construct/builder (get_turf(T.loc))
-						Z.key = A.key
-						Z.faction |= "\ref[U]"
-						if(iscultist(U))
-							if(SSticker.mode.name == "cult")
-								SSticker.mode:add_cultist(Z.mind)
-							else
-								SSticker.mode.cult+=Z.mind
-							SSticker.mode.update_cult_icons_added(Z.mind)
-						qdel(T)
 						to_chat(Z, "<B>You are an Artificer. You are incredibly weak and fragile, but you are able to construct fortifications, use magic missile, repair allied constructs (by clicking on them), </B><I>and most important of all create new constructs</I><B> (Use your Artificer spell to summon a new construct shell and Summon Soulstone to create a new soulstone).</B>")
-						to_chat(Z, "<B>You are still bound to serve your creator, follow [U.p_their()] orders and help [U.p_them()] complete [U.p_their()] goals at all costs.</B>")
-						Z.cancel_camera()
-						qdel(C)
+						init_construct(Z,A,C,T)
 			else
 				to_chat(U, "<span class='danger'>Creation failed!</span>: The soul stone is empty! Go kill someone!")
 	return
 
+/proc/init_construct(var/mob/living/simple_animal/hostile/construct/Z, var/mob/living/simple_animal/shade/A, var/obj/item/soulstone/C, var/obj/structure/constructshell/T)

```

Comment:
I hope you agree that variable names are completely random here and don't make any sense.
(I understand that changing this in a nice way may involve renaming all the variables throughout this file, so I'm not actually asking you to do this).

Reply:
Heck I'll look into it.

Reply:
Well I changed that proc and the "Construct" transfer soul to have slightly more descriptive letters. I'm not up for a full comb-over of the file. If Maints really want me to, I'll hecking do it.